libs/wingsdk/src/cloud/bucket.ts

```diff
@@ -23,6 +23,17 @@ export interface BucketProps {
   readonly public?: boolean;
 }
 
+/**
+ * Options for `Bucket.put`.
+ */
+export interface BucketPutProps {
+  /**
+   * The HTTP Content-Type of the object.
+   * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type
+   */
+  readonly contentType: string;
```

Comment:
This should be optional and have a default of auto-detecting the content type from the file extension.

Reply:
I checked the AWS docs and only the Java SDK has the built-in Content-Type autodetection (see [HERE](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/ObjectMetadata.html#setContentType-java.lang.String-)). So I guess you want to implement this functionality using the [mime module](https://www.npmjs.com/package/mime)?

Reply:
If the user takes a `.jpg` file and renames it to `.txt`, should the autodetection be able to determine the actual file type based on its binary information or should it just perform a shallow check looking only at its extension?

Reply:
The problem with `mime` module is that it determines the MIME type based solely on the file extension, and it doesn't examine the file's binary contents.

Reply:
> If the user takes a .jpg file and renames it to .txt, should the autodetection be able to determine the actual file type based on its binary information or should it just perform a shallow check looking only at its extension?

Just the extension. I don't see this as an issue. It's a very common practice to use the extension to determine the mime type.